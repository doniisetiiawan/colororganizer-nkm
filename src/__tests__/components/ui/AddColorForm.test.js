/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import { compose } from 'redux';
import AddColorFor from '../../../components/ui/AddColorForm';

configure({ adapter: new Adapter() });

describe('<AddColorForm /> UI Componnet', () => {
  let wrapper;

  describe('Rendering the UI', () => {
    it('renders correctly', () => {
      compose(
        expect,
        toJSON,
        shallow,
      )(<AddColorFor />).toMatchSnapshot();
    });

    it('Submitting default onNewColor does not cause error', () => {
      wrapper = mount(<AddColorFor />);
      wrapper.find('#_title').instance().value = 'test-color';
      wrapper.find('#_color').instance().value = '#0000FF';
      wrapper.find('form').simulate('submit');
    });
  });

  describe('Adding New Colors', () => {
    const _addColor = jest.fn();

    beforeAll(() => {
      wrapper = mount(
        <AddColorFor onNewColor={_addColor} />,
      );
      wrapper.find('#_title').instance().value = 'test-color';
      wrapper.find('#_color').instance().value = '#000099';
      wrapper.find('form').simulate('submit');
    });

    it('adds colors correctly', () => expect(_addColor).toBeCalledWith(
      'test-color',
      '#000099',
    ));

    it('resets the title value', () => expect(wrapper.find('#_title').instance().value).toBe(
      '',
    ));

    it('resets the color value', () => expect(wrapper.find('#_color').instance().value).toBe(
      '#000000',
    ));
  });
});
