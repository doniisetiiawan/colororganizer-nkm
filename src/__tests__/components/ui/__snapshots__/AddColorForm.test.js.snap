// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddColorForm /> UI Componnet Rendering the UI renders correctly 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      placeholder="color title..."
      required={true}
      type="text"
    />
    <input
      required={true}
      type="color"
    />
    <button
      type="submit"
    >
      ADD
    </button>
  </form>
</Fragment>
`;
